exercises/js/specials.js

```diff
@@ -0,0 +1,78 @@
+// Append a target div after the form that's inside the #specials element; this will be where you put information about the special once you receive it.
+// Bind to the change event of the select element; when the user changes the selection, send an Ajax request to /exercises/data/specials.json.
+
+// When the request returns a response, use the value the user selected in the select (hint: $.fn.val) to look up information about the special in the JSON response.
+// Add some HTML about the special to the target div you created.
+// Finally, because the form is now Ajax-enabled, remove the submit button from the form.
+// Note: that we're loading the JSON every time the user changes their selection. How could we change the code so we only make the request once, and then use a cached response when the user changes their choice in the select?
+const DEFAULT_OPTIONS = { containerSelector: '[data-load-json-container="container"]',
+                          removeButton: "input[type='submit']" };
+class LoadJson {
+  constructor(options) {
+    this.options = Object.assign({}, DEFAULT_OPTIONS, options);
+    this.$container = $(this.options.containerSelector);
+    this.$form = this.$container.find("form");
+    this.$select = this.$form.find("select");
+  }
+
+  init() {
+    const $divContainer = $("<div></div>");
+    $divContainer.insertAfter(this.$form);
+    this.$select.data("content", $divContainer);
+    this.bindChangeEvent();
+    this.removeSubmitButton();
+  }
+
+  removeSubmitButton() {
+    this.$form.find(this.options.removeButton).remove();
+  }
+
+  bindChangeEvent() {
+    this.$select.bind("change", (event) => {
+      const val = this.$select.val();
+      this.setContentDivWithData(val);
+    });
+  }
+
+  setContentDivWithData(value) {
+    this.promiseToFetchDataService().then(resultData => {
+      this.setContentDiv(value);
+    }).catch(error => {
+      console.log(error);
+    });
+  }
+
+  promiseToFetchDataService() {
+    return new Promise((resolve, reject) => {
+      if(!this.data) {
```

Comment:
Subjectivity Alert❗

Some might want to do it like below ( not a rejection point though )

```js
if (this.data) {
  return resolve(this.data);  
}
// Logic of `fetchSpecialsData` here...instead of separate instance method.
// Since there is no other invocation of fetchSpecialsData from anywhere.
```